src/main/java/org/jenkinsci/plugins/googleplayandroidpublisher/TrackPublisherTask.java

```diff
@@ -44,35 +43,37 @@
      * @param rolloutFraction The rollout fraction, if track is a staged rollout.
      */
     protected void assignApksToTrack(Collection<Integer> versionCodes, ReleaseTrack track,
-            double rolloutFraction) throws IOException, UploadException {
+            double rolloutFraction, TrackRelease release) throws IOException, UploadException {
         // Determine which version codes should be unassigned
         final int newestVersionCodeAllowed = getNewestVersionCodeAllowed(versionCodes);
 
         // Prepare to assign the APK(s) to the desired track
-        final Track trackToAssign = new Track();
-        trackToAssign.setTrack(track.getApiValue());
-        trackToAssign.setVersionCodes(new ArrayList<Integer>(versionCodes));
+        final Track trackToAssign = new Track()
+                .setTrack(track.getApiValue())
+                .setReleases(Collections.singletonList(release));
+
         if (track == PRODUCTION) {
             // Remove older APKs from the beta track
             unassignOlderApks(BETA, newestVersionCodeAllowed);
 
             // If there's an existing rollout, we need to clear it out so a new production/rollout APK can be added
             final Track rolloutTrack = fetchTrack(ROLLOUT);
-            if (rolloutTrack != null) {
+            double currentFraction = rolloutFraction;
+
+            if (rolloutTrack != null && rolloutTrack.getReleases() != null && !rolloutTrack.getReleases().isEmpty()) {
+                currentFraction = rolloutTrack.getReleases().get(0).getUserFraction();
                 logger.println(String.format("Removing existing staged rollout APK(s): %s",
-                        join(rolloutTrack.getVersionCodes(), ", ")));
-                rolloutTrack.setVersionCodes(null);
+                        join(rolloutTrack.getReleases().get(0).getVersionCodes(), "")));
```

Comment:
Was it intentional to change the `", "` join parameter to an empty string?

i.e. when there are multiple APKs in the existing release, I assume this would now just concatenate the versionCodes together when printing this message.

Reply:
Hmm, no that does not look to be intentional. I think I was looking at the plugin's uses of Releases (where there will only be one anyway), or I wasn't able to find a way to get multiple APK versionCodes to be in staged rollout simultaneously.